### ISSUE TITLE
pretrained model still need compute_bn_statistics

### ISSUE THREAD

**Issue by weiliu620** [alexgkendall/caffe-segnet#48]
I'm wondering if I use pretrained model, do I need to correct for batchNorm? One thing I found is, if I use pretrained model, but don't do the compute_bn_statistics step, the model output different results when I run same input data twice. Is it related to batchnorm?

Thanks,

**Comment by Soham0** [comment]
I have the same question.

**Comment by alexgkendall** [comment]
The models in the segnet model zoo have had global dataset statistics applied to their BN layers - so you don't need to run compute_bn_statistics step. They are good to go! As long as you use the inference prototxt, then these models should be deterministic.

Unless of course you are using the bayesian model, then each forward pass will sample a different dropout mask.

Cheers
